src/cache_guttering.cpp

```diff
@@ -9,180 +10,208 @@ inline static node_id_t extract_left_bits(node_id_t number, int pos) {
 
 void CacheGuttering::print_r_to_l(node_id_t src) {
   std::cout << "src: " << src;
-  std::cout << "->" << extract_left_bits(src, l1_pos);
-  std::cout << "->" << extract_left_bits(src, l2_pos);
-  std::cout << "->" << extract_left_bits(src, l3_pos);
-  std::cout << "->" << extract_left_bits(src, RAM1_pos);
+  std::cout << "->(L1)" << extract_left_bits(src, level1_pos);
+  std::cout << "->(L2)" << extract_left_bits(src, level2_pos);
+  std::cout << "->(L3)" << extract_left_bits(src, level3_pos);
+  if (level4_gutters)
+    std::cout << "->(RM)" << extract_left_bits(src, level4_pos);
   std::cout << std::endl;
 }
```

Comment:
This is a good debugging function; I think we need more like it. Especially to print out fanout, elements per child, buffersizes, etc.